controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -443,6 +448,158 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_WSC(Socket *sd,
     return true;
 }
 
+bool master_thread::handle_cmdu_1905_channel_preference_report_message(
+    Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    LOG(INFO) << "received CHANNEL_PREFERENCE_REPORT_MESSAGE";
+
+    // TODO: in actual channel selection task, it is important to validate that rx mid is identical
+    // to the mid sent in channel preference request message
+    // auto mid = cmdu_rx.getMessageId();
+
+    // build channel request message
+    auto cmdu_tx_header =
+        cmdu_tx.create(0, ieee1905_1::eMessageType::CHANNEL_SELECTION_REQUEST_MESSAGE);
+
+    if (!cmdu_tx_header) {
+        LOG(ERROR) << "cmdu creation of type CHANNEL_SELECTION_REQUEST_MESSAGE, has failed";
+        return false;
+    }
+
+    // Define ruid list in order to create only one reply of tlvChannelPreference per ruid
+    std::list<sMacAddr> ruid_list;
```

Comment:
I don't see anywhere that you insert something to this list, I must be missing something - please enlighten me..

Reply:
You are correct, I forgot to push it. fixed